src/main/java/org/cryptomator/cryptofs/health/shortened/TrailingNullBytesInNameFile.java

```diff
@@ -0,0 +1,44 @@
+package org.cryptomator.cryptofs.health.shortened;
+
+import org.cryptomator.cryptofs.VaultConfig;
+import org.cryptomator.cryptofs.health.api.CommonDetailKeys;
+import org.cryptomator.cryptofs.health.api.DiagnosticResult;
+import org.cryptomator.cryptolib.api.Cryptor;
+import org.cryptomator.cryptolib.api.Masterkey;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Map;
+
+/**
+ * Result and fix for bug https://github.com/cryptomator/cryptofs/issues/121
+ */
+public class TrailingNullBytesInNameFile implements DiagnosticResult {
```

Comment:
Instead of just null, we might want to report (and fix) _any_ non-printable/whitespace char (ASCII < 0x20). While we don't know of a bug that adds such chars, it is certainly useful if such content is added by unknown causes and requires exactly the same fix.

Reply:
Considered for db83548. All trailing bytes after `.c9r` are removed.